src/test/javascript/spec/component/overview/course-discussion/course-discussion.component.spec.ts

```diff
@@ -0,0 +1,373 @@
+import { PostCreateEditModalComponent } from 'app/shared/metis/postings-create-edit-modal/post-create-edit-modal/post-create-edit-modal.component';
+import { ButtonComponent } from 'app/shared/components/button.component';
+import { FormBuilder, FormsModule, ReactiveFormsModule } from '@angular/forms';
+import { ActivatedRoute } from '@angular/router';
+import { HttpResponse } from '@angular/common/http';
+import { Observable, of } from 'rxjs';
+import { getElement } from '../../../helpers/utils/general.utils';
+import { PageDiscussionSectionComponent } from 'app/overview/page-discussion-section/page-discussion-section.component';
+import * as moment from 'moment';
+import { NgbTooltip } from '@ng-bootstrap/ng-bootstrap';
+import { MockComponent, MockDirective, MockModule, MockPipe } from 'ng-mocks';
+import { Course } from 'app/entities/course.model';
+import { CourseManagementService } from 'app/course/manage/course-management.service';
+import { ArtemisTranslatePipe } from 'app/shared/pipes/artemis-translate.pipe';
+import { MetisService } from 'app/shared/metis/metis.service';
+import { ExerciseService } from 'app/exercises/shared/exercise/exercise.service';
+import { MockExerciseService } from '../../../helpers/mocks/service/mock-exercise.service';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { ArtemisTestModule } from '../../../test.module';
+import { AnswerPostService } from 'app/shared/metis/answer-post.service';
+import { MockAnswerPostService } from '../../../helpers/mocks/service/mock-answer-post.service';
+import { PostService } from 'app/shared/metis/post.service';
+import { MockPostService } from '../../../helpers/mocks/service/mock-post.service';
+import { AccountService } from 'app/core/auth/account.service';
+import { MockAccountService } from '../../../helpers/mocks/service/mock-account.service';
+import { CourseDiscussionComponent } from 'app/overview/course-discussion/course-discussion.component';
+import {
+    metisAnswerPostUser1,
+    metisCourse,
+    metisCoursePosts,
+    metisCoursePostsWithCourseWideContext,
+    metisExercise,
+    metisExercisePosts,
+    metisLecture,
+    metisLecturePosts,
+    metisPostExerciseUser1,
+    metisPostExerciseUser2,
+    metisPostLectureUser1,
+    metisPostLectureUser2,
+    metisUpVoteReactionUser1,
+} from '../../../helpers/sample/metis-sample-data';
+
+chai.use(sinonChai);
+const expect = chai.expect;
+
+describe('CourseDiscussionComponent', () => {
+    let component: CourseDiscussionComponent;
+    let fixture: ComponentFixture<CourseDiscussionComponent>;
+    let courseManagementService: CourseManagementService;
+    let metisService: MetisService;
+    let metisServiceSpy: SinonSpy;
+    let post1: Post;
+    let post2: Post;
+    let post3: Post;
+    let post4: Post;
+    let posts: Post[];
+
+    beforeEach(() => {
+        const id = metisCourse.id;
```

Comment:
I would perhaps make sense to define them in the outer scope, this way they get redefined each test